### PR TITLE
[beken-72xx] Add IPv6 and lwIP 2.2.0 support

### PR DESCRIPTION
**Pull request by dwmw2**
~~Once https://github.com/libretiny-eu/lwip/pull/2 is merged to a new 2.2.0-bdk branch, that should be added to platform.json too.~~

### CODE REVIEW

File: cores/beken-72xx/arduino/libraries/WiFi/WiFiSTA.cpp [hunk 1]
```diff
@@ -246,3 +246,39 @@ WiFiAuthMode WiFiClass::getEncryption() {
...
+	std::vector<IPv6Address> result;
+	struct wlan_ip_config addr;
+	int nr_addresses = 0;
+
+	if (sta_ip_is_start())
+		nr_addresses = net_get_if_ipv6_pref_addr(&addr, ifs);
+
+	for (int i = 0; i < nr_addresses; i++)
+		if (ip6_addr_islinklocal(&addr.ipv6[i]))
+			return IPv6Address(addr.ipv6[i].addr);
```
Comment (kuba2k2) [inline cores/beken-72xx/arduino/libraries/WiFi/WiFiSTA.cpp:265]: I'd prefer having braces around loops:

```suggestion
	for (int i = 0; i < nr_addresses; i++) {
		if (ip6_addr_islinklocal(&addr.ipv6[i]))
			return IPv6Address(addr.ipv6[i].addr);
	}
```

File: cores/beken-72xx/arduino/libraries/WiFi/WiFiSTA.cpp [hunk 2]
```diff
@@ -246,3 +246,39 @@ WiFiAuthMode WiFiClass::getEncryption() {
...
+	struct netif *ifs = (struct netif *)net_get_sta_handle();
+	std::vector<IPv6Address> result;
+	struct wlan_ip_config addr;
+	int nr_addresses = 0;
+
+	if (sta_ip_is_start())
+		nr_addresses = net_get_if_ipv6_pref_addr(&addr, ifs);
+
+	for (int i = 0; i < nr_addresses; i++)
+		result.push_back(IPv6Address(addr.ipv6[i].addr));
```
Comment (kuba2k2) [inline cores/beken-72xx/arduino/libraries/WiFi/WiFiSTA.cpp:280]:
```suggestion
	for (int i = 0; i < nr_addresses; i++) {
		result.push_back(IPv6Address(addr.ipv6[i].addr));
	}
```

File: cores/common/arduino/libraries/common/mDNS/LwIPmDNS.cpp [hunk 3]
```diff
@@ -73,10 +73,15 @@ static void mdnsTxtCallback(struct mdns_service *service, void *userdata) {
...
 static void mdnsStatusCallback(struct netif *netif, uint8_t result) {
 	LT_DM(MDNS, "Status: netif %u, status %u", netif->num, result);
 }
-
+#else
+static void mdnsStatusCallback(struct netif *netif, uint8_t result, int8_t slot) {
+	LT_DM(MDNS, "Status: netif %u, status %u slot %d", netif->num, result, slot);
+}
+#endif
 #ifdef LWIP_NETIF_EXT_STATUS_CALLBACK
```
Comment (kuba2k2) [inline cores/common/arduino/libraries/common/mDNS/LwIPmDNS.cpp:85]:
```suggestion

#ifdef LWIP_NETIF_EXT_STATUS_CALLBACK
```

**Review by kuba2k2 — CHANGES_REQUESTED** [review]

File: external-libs.json [hunk 4]
```diff
@@ -90,9 +90,10 @@
 			"+<src/api/*.c>",
 			"+<src/core/*.c>",
 			"+<src/core/ipv4/*.c>",
+			"+<src/core/ipv6/*.c>",
```
[the remark this answers is not in the document]
Comment (dwmw2) [reply external-libs.json:93]: (Note: Strictly, updating to 2.2.0 probably isn't needed for IPv6)

**Review by kuba2k2 — APPROVED** [review]

**Pull request merged**